node/mock-task.ts

```diff
@@ -407,11 +407,13 @@ export function matchFile(list, pattern, options): string[] {
     return mock.getResponse('match', pattern) || [];
 }
 
-export function filter(pattern, options): any {
-    var filterList = mock.getResponse('filter', pattern) || [];
-	return function(pattern, i, arr) {
-		return filterList.indexOf(pattern) >= 0;
-	}
+export function filter(pattern, options): (element: string, index: number, array: string[]) => boolean {
```

Comment:
On the one hand, I really like the idea of functions like `filter` (and `match`) being able to use the real behavior. However, what feels strange to me about this implementation is, *some* `filter` calls may be mocked, and others use the real functionality.

I think an all-or-nothing behavior would be more inline with common mocking approaches. That is, only "fallback" if 'filter' isn't registered at all in the answer object. That would mean changing MockAnswer to expose a function like `isRegistered(cmd: string): boolean`.

Thgouhts?